src/panels/config/entity_registry/ha-config-entity-registry.ts

```diff
@@ -123,16 +123,28 @@ class HaConfigEntityRegistry extends LitElement {
 
   private _openEditEntry(ev: MouseEvent): void {
     const entry = (ev.currentTarget! as any).entry;
+    const entryName = computeEntityRegistryName(this.hass!, entry);
+    const entityIdPattern = entry.entity_id.split(".").pop();
+
+    const matchingEntities = this._items.filter((e) => {
```

Comment:
Instead of matching like this, we should find all entitty registry entries that have the same device_id as the one picked.